### PR TITLE
improve logging for creating and patching conferences through the RES…

### PR DESCRIPTION
**Pull request by davidertel**
…T inteface. Return an Error header with more information about why the request failed. This helps debug failures that occur when using the REST API.

### CODE REVIEW

**Comment by davidertel** [comment]
Adding the error header is just one way of reporting the errors back. Currently all of the errors are quietly swallowed through the REST API.

**Comment by davidertel** [comment]
any comments or suggestions on this PR?

**Comment by bbaldino** [comment]
looks good to me and this would give some good feedback when using the rest api... @lyubomir any thoughts?

**Comment by bgrozev** [comment]
Just an opinion: for health checks (/about/health) we include similar information in the content (if the response code is 4xx/5xx). For consistency, I'd prefer to use the same approach in both places, whether that's using the content or a header field.

**Comment by davidertel** [comment]
I'll update to put the content in the body and remove the header to be consistent with the health check controller

**Comment by davidertel** [comment]
Moved the error messages to the response body as requested, keeps it consistent with the health check

File: src/main/java/org/jitsi/videobridge/rest/HandlerImpl.java [hunk 1]
```diff
@@ -667,16 +670,18 @@ else if (RESTUtil.isJSONContentType(request.getContentType()))
                         }
                     }
                 }
-                if (status != 0)
+                else {
                     response.setStatus(status);
             }
```
Comment (bbaldino) [inline src/main/java/org/jitsi/videobridge/rest/HandlerImpl.java]: mis-aligned brace here?  also the opening brace for the else should be on a newline

**Comment by bbaldino** [comment]
lgtm.  @bgrozev ?

**Comment by bgrozev** [comment]
No objections here

**Pull request merged**